File: sdk/ruby/test/fixtures/run_worker_stdio.rb
#!/usr/bin/env ruby
# frozen_string_literal: true

# Same worker as `run_worker.rb`, over the framed stdio fallback transport.

here = File.expand_path(__dir__)
$LOAD_PATH.unshift(here)
$LOAD_PATH.unshift(File.expand_path('../../lib', here))

require 'portals'
require 'bench_worker'
require 'conformance_worker'

Portals::Worker.new(max_concurrency: 16, max_streams: 8, transport: :stdio).run

File: sdk/node/test/fixtures/run_worker_stdio.js
#!/usr/bin/env node
'use strict';

// Same worker as `run_worker.js`, over the framed stdio fallback transport.

const { Worker } = require('../..');

new Worker({
  maxConcurrency: 16,
  maxStreams: 8,
  transport: 'stdio',
  modules: {
    bench_worker: require('./bench_worker'),
    conformance_worker: require('./conformance_worker'),
  },
  modulePaths: [__dirname],
}).run().catch((err) => {
  process.stderr.write(`portals worker: ${err.stack || err}\n`);
  process.exit(1);
});
